src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs

```diff
@@ -131,10 +131,72 @@ public PackageArchiveReader BuildPackage(PackageBuilder builder, string outputPa
                 PrintVerbose(outputPath, builder);
             }
 
+            if (_packArgs.InstallPackageToOutputPath)
+            {
+                _packArgs.Logger.LogMinimal(string.Format(CultureInfo.CurrentCulture, Strings.Log_PackageCommandInstallPackageToOutputPath, "Package", outputPath));
+                WriteResolvedNuSpecToPackageOutputDirectory(builder);
+                WriteSHA512PackageHash(builder);
+            }
+
             _packArgs.Logger.LogMinimal(String.Format(CultureInfo.CurrentCulture, Strings.Log_PackageCommandSuccess, outputPath));
             return new PackageArchiveReader(outputPath);
         }
 
+        /// <summary>
+        /// Writes the resolved NuSpec file to the package output directory.
+        /// </summary>
+        /// <param name="builder">The package builder</param>
+        private void WriteResolvedNuSpecToPackageOutputDirectory(PackageBuilder builder)
+        {
+            var outputPath = GetOutputPath(builder, _packArgs, false, builder.Version);
+            Directory.CreateDirectory(Path.GetDirectoryName(outputPath));
+
+            var resolvedNuSpecOutputPath = Path.Combine(
+                Path.GetDirectoryName(outputPath), 
+                new VersionFolderPathResolver(outputPath).GetManifestFileName(builder.Id, builder.Version));
+
+            _packArgs.Logger.LogMinimal(string.Format(CultureInfo.CurrentCulture, Strings.Log_PackageCommandInstallPackageToOutputPath, "NuSpec", resolvedNuSpecOutputPath));
+
+            if (string.Equals(_packArgs.Path, resolvedNuSpecOutputPath, StringComparison.OrdinalIgnoreCase))
+            {
+                throw new PackagingException(NuGetLogCode.NU5001, string.Format(CultureInfo.CurrentCulture, Strings.Error_WriteResolvedNuSpecOverwriteOriginal, _packArgs.Path));
+            }
+
+            var manifest = new Manifest(new ManifestMetadata(builder), null);
+            var tempOutputPath = Path.Combine(NuGetEnvironment.GetFolderPath(NuGetFolderPath.Temp), Path.GetFileName(resolvedNuSpecOutputPath));
```

Comment:
I cannot have a temp file with a guid getting dropped into the same directory (var fileName = $"{Guid.NewGuid()}.tmp".ToLowerInvariant();). This would break caching as we have to declare what the output file names are before we run the nuget pack command and any file that shows up during building that is unexpected fails the build. This line in GetTempFilePath would create a temp file name with a guid in the same directory as the package. However, we can get away with creating a temp file name in NuGetEnvironment.GetFolderPath(NuGetFolderPath.Temp) because this folder is a special case in the build system as many tools use it (it is based on Path.GetTempPath()).